### ISSUE TITLE
L>M Upgrade: Ansible roles are downloaded twices

### ISSUE THREAD

**Issue by BjoernT** [rcbops/rpc-openstack#1524]
###### Please provide the output to 'git describe --all' for  all deployments where you have confirmed it occurs.

r13.1.0rc2
13.3.4-1-ga489036

##### What type of issue is this? Please tick the appropriate box.
- [X] bug - existing functionality that does not work for you as documented
- [ ] feature - new functionality that you would like to see added

##### What is the expected behavior?

We download OSA and RPCO ansible roles only once in one step

##### What is actually happening?

The OSA bootstrap-ansible.sh script is downloading the roles and then later the RPC-O upgrade too via the documented step

```
ansible-galaxy install --role-file=/opt/rpc-openstack/openstack-ansible/ansible-role-requirements.yml --force \
        --roles-path=/opt/rpc-openstack/rpcd/playbooks/roles )
```
##### What were you attempting to do when you encountered this error (give as much detail as possible)? 

N/A

##### How can we reproduce this issue?

Execute the upgrade from L to M 

##### How can we verify your issue is addressed?

No redundant download is happening

##### If you are using or know of a workaround please describe the necessary steps?

N/A

##### What is your evaluation on the importance of this issue, and why?

Pio 3

----------------------------------------------------------------------------------------------------------------------------
###### For each pull request, update the branch line to reference it.
- [ ] master
- [ ] mitaka-13.1

##### Upgrade impact. Consider the impact on moving between major, minor and patch releases.
- [ ] This change does not impact upgrades
- [ ] This change addresses a potential upgrade impact
- [ ] The upgrade impact is being tracked by another issue.

The issue tracking the upgrade impact is:

##### Testing impact
- [ ] An issue/pull-request has been create or exists to track the testing gap, and is referenced below
- [ ] This change includes additional testing
- [ ] This change is tested by the existing setup

Additional testing links:

##### Release note
- [ ] This change includes a release note.
- [ ] This change does not require a release note.

##### Does your change require documentation?
- [ ] Yes, and I submitted an issue to [Docs issues](https://github.com/rackerlabs/docs-rpc/issues "Docs issues")
- [ ] Yes, and I contributed a documentation patch to [docs-rpc](https://github.com/rackerlabs/docs-rpc "docs-rpc")
- [ ] No documentation required for this change

Additional documentation links (please add links if you have ticked yes):

**Comment by alextricity25** [comment]
Hi @BjoernT.
The command you pasted in the description is:
```
ansible-galaxy install **--role-file=/opt/rpc-openstack/openstack-ansible/ansible-role-requirements.yml** --force \
        --roles-path=/opt/rpc-openstack/rpcd/playbooks/roles )
```
But the command [documented](https://github.com/rackerlabs/docs-rpc/blob/master/doc/rpc-upgrade/rpc-upgrade-v12-v13-perform.rst#update-the-rpc-openstack-galaxy-modules) is:
```
ansible-galaxy install \
**--role-file=/opt/rpc-openstack/ansible-role-requirements.yml \**
--force --roles-path=/opt/rpc-openstack/rpcd/playbooks/roles
```
Notice how the first uses ``--role-file=/opt/rpc-openstack/openstack-ansible/ansible-role-requirements.yml``, while the second one uses ``--role-file=/opt/rpc-openstack/ansible-role-requirements.yml``(The requirements file in the rpc-openstack repo, not openstack-ansible). The latter will re-download the ceph roles listed [here](https://github.com/rcbops/rpc-openstack/blob/master/ansible-role-requirements.yml). This is so the new ceph role code base is pulled in.

Hopefully that explains and addresses any concerns and I'm not missing anything obvious here.

**Comment by BjoernT** [comment]
OK my bad that seems good enough for now

**Comment by claco** [comment]
@BjoernT Not sure if intentional, or accidental, but I read "twices" and laughed.  :-)

**Comment by BjoernT** [comment]
Lol

**Comment by odyssey4me** [comment]
Note that it may be useful to set the OSA ansible bootstrap to process the ansible-role-requirements as a list of files rather than a single file. If this is done then the RPC process could be made a lot simpler and require less manual steps.
